web/apps/apps_test.go

```diff
@@ -0,0 +1,147 @@
+package apps
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/cozy/cozy-stack/apps"
+	"github.com/cozy/cozy-stack/config"
+	"github.com/cozy/cozy-stack/couchdb"
+	"github.com/cozy/cozy-stack/instance"
+	"github.com/cozy/cozy-stack/vfs"
+	"github.com/cozy/cozy-stack/web/middlewares"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+const domain = "cozy-with-apps.example.net"
+const slug = "mini"
+
+var ts *httptest.Server
+var testInstance *instance.Instance
+
+func injectInstanceAndSlug(i *instance.Instance, s string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		c.Set("instance", i)
+		c.Set("app_slug", s)
+	}
+}
+
+func createFile(dir, filename, content string) error {
+	abs := path.Join(dir, filename)
+	file, err := vfs.Create(testInstance, abs)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	file.Write([]byte(content))
+	return nil
+}
+
+func installMiniApp() error {
+	man := &apps.Manifest{
+		Slug:   slug,
+		Source: "git://github.com/cozy/mini.git",
+		State:  apps.Ready,
+	}
+
+	err := couchdb.CreateNamedDoc(testInstance, man)
+	if err != nil {
+		return err
+	}
+
+	appdir := path.Join(apps.AppsDirectory, slug)
+	err = vfs.MkdirAll(testInstance, appdir)
+	if err != nil {
+		return err
+	}
+	pubdir := path.Join(appdir, "public")
+	err = vfs.Mkdir(testInstance, pubdir)
+	if err != nil {
+		return err
+	}
+
+	err = createFile(appdir, "index.html", "this is index.html")
+	if err != nil {
+		return err
+	}
+	err = createFile(appdir, "hello.html", "world")
+	if err != nil {
+		return err
+	}
+	err = createFile(pubdir, "index.html", "this is a file in public/")
+	if err != nil {
```

Comment:
gometalinter & travis dislike this condition (if err != nil { return err }; return nil)